src/init.sh

```diff
@@ -170,5 +180,173 @@ function init_help()
+# for future access. Checks if these values are empty and store this information
+# in a boolean variable inside the array.
+#
+# @git_config: associative array received as a nameref attribute to store all
+#              values acquired in this function.
+#
+# Returns: git_config[configured]: 1, if there is any revelant information in
+#          git missing, and 0, if git is fully configured.
+function get_git_config()
+{
+  local -n git_config="$1"
+
+  git_config['name']=$(git config user.name)
+  git_config['email']=$(git config user.email)
+  git_config['editor']=$(git config core.editor)
+  git_config['branch']=$(git config init.defaultBranch)
+
+  git_config['configured']=0
+  if [[ -z "${git_config['name']}" ||
+    -z "${git_config['email']}" ||
+    -z "${git_config['editor']}" ||
+    -z "${git_config['branch']}" ]]; then
+    git_config['configured']=1
+  fi
+}
+
+# This function sets if the scope of the git configuration will be
+# local or global.
+#
+# @config_cmd: command in which will be added the configuration scope.
+#
+# Returns: the git config command, adjusted if the scope is local or global
+function set_git_config_scope()
+{
+  local command
+  local scope
+
+  command='git config'
+
+  if is_inside_work_tree; then
+    printf '\nSelect the scope of this configuration:\n' 1>&2
+
+    select scope in 'Local' 'Global'; do
+      case $scope in
+        'Global')
+          command+=' --global'
+          break
+          ;;
+        'Local')
+          command+=' --local'
+          break
+          ;;
+      esac
+    done
+  else
+    warning 'You are not in a git repository. All modifications will be global' 1>&2
+    if [[ $(ask_yN 'Would you like to proceed?' 'y') =~ '1' ]]; then
+      command+=' --global'
+    else
+      command=''
+    fi
+  fi
+
+  printf '%s\n' "$command"
+}
+
+# Allows the user to set the empty configurations if any.
+#
+# @config_git: associative array with data about git configuration.
+#
+# Returns: config_git: array with the selected information configured on git
+function set_git_config()
+{
+  local -n config_git="$1"
+  local flag="$2"
+  local git_editor_suggestion
+  local config_cmd
+  local scope
+
+  flag=${flag:-'SILENT'}
+
+  if [[ "${config_git['configured']}" =~ '1' ]]; then
+    printf '\nNow Git will be configured.\n' 1>&2
+
+    config_cmd=$(set_git_config_scope)
+    if [[ -z $config_cmd ]]; then
+      return
+    fi
+
+    if [[ -z "${config_git['name']}" ]]; then
+      if [[ $(ask_yN $'\nWould you like to configure your name on Git?' 'y') =~ '1' ]]; then
+        config_git['name']=$(ask_with_default 'What is your name?' "$USER")
+        cmd_manager "$flag" "$config_cmd user.name \"${config_git['name']}\""
+      fi
+    fi
+
+    if [[ -z "${config_git['email']}" ]]; then
+      if [[ $(ask_yN $'\nWould you like to configure your email on Git?' 'y') =~ '1' ]]; then
+        while :; do
+          printf 'Insert nothing to cancel this.\n' 1>&2
```

Comment:
Should we allow cancellation here? I think we should enforce uses to set an email.